phageAPI/populate.py

```diff
@@ -1,19 +1,79 @@
+#!/usr/bin/env python
+
 import os
-import fetch
-from IPython import embed
-from Bio import SeqIO
-from lxml import html, etree
+import sys
+import argparse
 import requests
 import pandas
 import pickle
-from prunedict import prunedict
+from lxml import html, etree
+from Bio import Entrez
+from Bio import SeqIO
 from tqdm import tqdm
 
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'phageAPI.settings')
+import django
+django.setup()
+
+from util.acc import read_accession_file
+from util.prunedict import prunedict
+from util import fetch
+from restapi.models import (
+    Organism,
+    Spacer,
+    Repeat,
+    OrganismSpacerRepeatPair,
+    AntiCRISPR
+)
+
+DATA_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), '../data'))
+
+
+def status(count, done=False):
+    sys.stdout.write("\rCount: %9i" % count);
+    if done:
+        sys.stdout.write("\n")
+
+
+def populate_organism():
+    def add_organism(name, accession):
+        # get the object, this also checks for duplicates
+        o, created = Organism.objects.get_or_create(name=name, accession=accession)
+        return o
+
+    def merge_acc_names(accession_list):
+        acc_name_dict = {}
+        db = "nuccore"
+        count = 0
+        # Doing batches of 200 to make sure requests to NCBI are not too big
+        for i in range(0, len(accession_list), 200):
+            j = i + 200
+            if (j >= len(accession_list)):
```

Comment:
NCBI recommended maximum ids for a single request is 200 but actually there's a better way which uses HTTP POST to give the whole list of ids first and then use the session id to get the entries.